skiko/src/commonMain/kotlin/org/jetbrains/skia/impl/Native.kt

```diff
@@ -109,8 +109,8 @@ inline fun withResult(result: NativePointerArray, block: (InteropPointer) -> Uni
 }
 
 @Suppress("NON_PUBLIC_CALL_FROM_PUBLIC_INLINE")
-inline fun withStringResult(block: () -> NativePointer): String {
-    val string = ManagedString(block())
+inline fun withStringResult(managed: Boolean = true, block: () -> NativePointer): String {
```

Comment:
Why would we want `managed = false` here?

Reply:
Some SkString instances are not owned by another objects, we are receiving them by const&. So it is UB to delete them in ManagedString

Reply:
For such cases we shall hold reference to those objects, as in here https://github.com/JetBrains/skiko/blob/09aad6044e04a0b87ae4a9489de9c2236417d735/skiko/src/commonMain/kotlin/org/jetbrains/skia/Pixmap.kt#L12

Reply:
This requires a special Managed class just to keep temporary objects. IMHO it is an overkill